PushRelabelService.cpp

```diff
@@ -0,0 +1,104 @@
+#include "PushRelabelService.h"
+
+PushRelabelService::PushRelabelService()
+{
+
+}
+
+PushRelabelService::~PushRelabelService()
+{
+
+}
+
+int PushRelabelService::calculateMaxFlow(Graph* graph, int source, int destination)
+{
+	Graph* residualGraph = graph->clone();
+	vector<int> queue;
+	int* e = new int[graph->getNumberOfVertices()] ();
+	int* h = new int[graph->getNumberOfVertices()] ();
+	bool* inQueue = new bool[graph->getNumberOfVertices()] ();
+
+	h[source] = graph->getNumberOfVertices();
+
+	for (int i = 0; i < graph->getNumberOfVertices(); i++)
+	{
+		if (graph->getAdjacencyMatrix()[source][i] == 0)
+		{
+			continue;
+		}
+		residualGraph->getAdjacencyMatrix()[source][i] = 0;
+		residualGraph->getAdjacencyMatrix()[i][source] = graph->getAdjacencyMatrix()[source][i];
+		e[i] = graph->getAdjacencyMatrix()[source][i];
+		if (i != destination)
+		{
+			queue.push_back(i);
+			inQueue[i] = true;
+		}
+	}
+
+	while (!queue.empty())
+	{
+		int u = queue[0];
+		queue.erase(queue.begin());
```

Comment:
Jeśli queue jest używane jako kolejka, to lepiej byłoby zdefiniować tę zmienną jako std::queue (ewentualnie std::list), a nie std::vector. Metoda std::vector::erase jest bardziej czasochłonna niż pobranie pierwszego elementu kolejki.

Reply:
w sumie w pewnym momencie na std::list sam trafiłem, ale nie chciało mi się już zmieniać xD. No ale w sumie racja, podmienię to.